src/component/MovieListPage/MovieListPage.js

```diff
@@ -133,29 +190,44 @@ function MovieListPage({ }) {
 
     //Used to set the currently selected movieId in the URL
     useEffect(() => {
-        if(selectedMovie != null) {
-            setParamsInURL(selectedMovie.id)
+        if (selectedMovie != null) {
+            setParamsInURL('', selectedMovie.id)
         }
     }, [selectedMovie])
 
-    function setParamsInURL(id) {
+    function setParamsInURL(url, id) {
         const params = new URLSearchParams();
-            if (searchQuery) params.set('query', searchQuery);
-            if (activeGenre) params.set('genre', activeGenre);
-            if (selectedFilter) params.set('sortBy', selectedFilter);
-            params.set('offset', offset.toString());
-            if (id) {
-                window.history.pushState({}, '', `/${id}?${params.toString()}`);
-            } else {
-                window.history.pushState({}, '', `/?${params.toString()}`);
-            }
+        if (searchQuery) params.set('query', searchQuery);
+        if (activeGenre) params.set('genre', activeGenre);
+        if (selectedFilter) params.set('sortBy', selectedFilter);
+        params.set('offset', offset.toString());
+        if (id) {
+            navigate(`${url}/${id}?${params.toString()}`);
+        } else {
+            navigate(`${url}?${params.toString()}`);
+        }
     }
     return (
         <>
             {selectedMovie == null ?
-                <SearchForm onSearch={handleSearch} /> :
+                <SearchForm onSearch={handleSearch} initialQuery={searchQuery} displayAddMovieDialog={handleAddMovieDialogChange} /> :
                 <MovieDetails {...selectedMovie} onSearchSelect={handleSelectedMovie} />
             }
+            {isAddMovieDialogVisible &&
+                <AddMovie displayAddMovieDialog={handleAddMovieDialogChange} />
+            }
+
+            {isEditMovieDialogVisible && movieToEdit && (
```

Comment:
If I were you, for more readability I would suggest
`const canUpdateMovie = isEditMovieDialogVisible && movieToEdit`

Reply:
Fixed as part of a5b5e33